src/gtc/passes/oir_optimizations/horizontal_execution_merging.py

```diff
@@ -199,18 +236,20 @@ def visit_VerticalLoop(self, node: oir.VerticalLoop, **kwargs: Any) -> oir.Verti
     def visit_Stencil(self, node: oir.Stencil, **kwargs: Any) -> oir.Stencil:
         vertical_loops: List[oir.VerticalLoop] = []
         protected_fields = set(n.name for n in node.params)
-        for vl in reversed(node.vertical_loops):
-            vertical_loops.insert(
-                0,
-                self.visit(
-                    vl,
-                    new_symbol_name=symbol_name_creator(set(kwargs["symtable"])),
-                    protected_fields=protected_fields,
-                    **kwargs,
-                ),
+        all_names = collect_symbol_names(node)
+        vertical_loops = [
+            *reversed(
+                [
+                    self.visit(
+                        vl,
+                        new_symbol_name=symbol_name_creator(all_names),
+                        protected_fields=protected_fields,
+                        **kwargs,
+                    )
+                    for vl in reversed(node.vertical_loops)
+                ]
             )
-            access_collection = AccessCollector.apply(vl)
-            protected_fields |= access_collection.fields()
+        ]
```

Comment:
Why are the tests passing if `protected_fields` is not longer updated inside the loop with the results of the `AccessCollector` pass? Either the rests are not detailed enough or the logic in the loop was previously wrong.

Reply:
Earlier the pass was using protected fields as a way of not creating temporaries that already existed as named fields. Now that's being done automatically in a different way.

Reply:
I did take note of that, but thanks for thinking of it!

Reply:
Ok, it makes sense. Thanks for the explanation.